base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -258,644 +317,657 @@ pub fn create_block_hashing_blob(
-            total_kernel_offset: Default::default(),
-            total_script_offset: Default::default(),
-            nonce: 0,
-            pow: ProofOfWork::default(),
-            validator_node_mr: FixedHash::zero(),
-            validator_node_size: 0,
-        };
-        let hash = block_header.merge_mining_hash();
-        insert_merge_mining_tag_into_block(&mut block, hash).unwrap();
-        let count = 1 + (u16::try_from(block.tx_hashes.len()).unwrap());
-        let mut hashes = Vec::with_capacity(count as usize);
-        let mut proof = Vec::with_capacity(count as usize);
-        hashes.push(block.miner_tx.hash());
-        proof.push(block.miner_tx.hash());
-        for item in block.clone().tx_hashes {
-            hashes.push(item);
-            proof.push(item);
-        }
-
-        let coinbase_merkle_proof = create_merkle_proof(&hashes).unwrap();
-        let monero_data = MoneroPowData {
-            header: block.header,
-            randomx_key: FixedByteArray::from_canonical_bytes(&from_hex(&seed_hash).unwrap()).unwrap(),
-            transaction_count: count,
-            merkle_root: Hash::null(),
-            coinbase_merkle_proof,
-            coinbase_tx: block.miner_tx,
-        };
-        let mut serialized = Vec::new();
-        monero_data.serialize(&mut serialized).unwrap();
-        let pow = ProofOfWork {
-            pow_algo: PowAlgorithm::RandomX,
-            pow_data: serialized,
-        };
-        block_header.pow = pow;
-        let err = verify_header(&block_header).unwrap_err();
-        unpack_enum!(MergeMineError::InvalidMerkleRoot = err);
-    }
-
-    #[test]
-    fn test_difficulty() {
-        // Taken from block: https://stagenet.xmrchain.net/search?value=672576
-        let versions = "0c0c";
-        // Tool for encoding VarInts:
-        // https://gchq.github.io/CyberChef/#recipe=VarInt_Encode()To_Hex('Space',0)From_Hex('Auto'/disabled)VarInt_Decode(/disabled)&input=MTYwMTAzMTIwMg
-        let timestamp = "a298b7fb05"; // 1601031202
-        let prev_block = "046f4fe371f9acdc27c377f4adee84e93b11f89246a74dd77f1bf0856141da5c";
-        let nonce = "FE394F12"; // 307182078
-        let tx_hash = "77139305ea53cfe95cf7235d2fed6fca477395b019b98060acdbc0f8fb0b8b92"; // miner tx
-        let count = "01";
-
-        let input = from_hex(&format!(
-            "{}{}{}{}{}{}",
-            versions, timestamp, prev_block, nonce, tx_hash, count
-        ))
-        .unwrap();
-        let key = from_hex("2aca6501719a5c7ab7d4acbc7cc5d277b57ad8c27c6830788c2d5a596308e5b1").unwrap();
-        let rx = RandomXFactory::default();
-
-        let (difficulty, hash) = get_random_x_difficulty(&input, &rx.create(&key).unwrap()).unwrap();
-        assert_eq!(
-            hash.to_hex(),
-            "f68fbc8cc85bde856cd1323e9f8e6f024483038d728835de2f8c014ff6260000"
-        );
-        assert_eq!(difficulty.as_u64(), 430603);
-    }
-}
+// #[cfg(test)]
+// mod test {
```

Comment:
I assume these tests will be updated?

Reply:
fudge, good catch, I forgot I disabled those, let me fix them